pkg/ottl/expression.go

```diff
@@ -245,6 +246,45 @@ func (g StandardFloatGetter[K]) Get(ctx context.Context, tCtx K) (float64, error
 	}
 }
 
+// FunctionGetter is a Getter that must return an ExprFunc.
+type FunctionGetter[K any] interface {
+	Get(ctx context.Context, iArgs Arguments) (Expr[K], error)
+}
+
+// StandardFunctionGetter is a basic implementation of FunctionGetter
+type StandardFunctionGetter[K any] struct {
+	fCtx FunctionContext
+	fact Factory[K]
+}
+
+// Get retrieves an ExprFunc value.
+// If the value is not an ExprFunc a new Error is returned.
+// If there is an error getting the value it will be returned.
+func (g StandardFunctionGetter[K]) Get(_ context.Context, iArgs Arguments) (Expr[K], error) {
+	if g.fact == nil {
+		return Expr[K]{}, fmt.Errorf("undefined function")
+	}
+	fArgs := g.fact.CreateDefaultArguments()
+	fArgsVal := reflect.ValueOf(fArgs).Elem()
+	iArgsVal := reflect.ValueOf(iArgs).Elem()
```

Comment:
We need to check whether these are pointer values or this will panic if they're not. See here for how we do this check during standard function instantiation: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/pkg/ottl/v0.82.0/pkg/ottl/functions.go#L32

Reply:
for some reason this condition passes

```if reflect.TypeOf(fArgs).Kind() != reflect.Pointer```

 reflect.TypeOf(fArgs).Kind() is of type `ptr`

Reply:
If you're testing with how our functions are currently defined, then I think it should always, yeah. However if you change something like `&stringGetterArguments{}` to `stringGetterArguments{}` on [this test](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24356/files#diff-3f8822d6a24dc5a9d3a52c3c72fdf45f1621e52d5e782b8576401cca2a913330R1673) I believe the code will panic as-written.

Reply:
I was able to add the check in the last [commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24356/commits/371c56501da93fc31321aeafa078bd2f9f12303b)

Reply:
I think this completes all the changes